### ISSUE TITLE
[On going...] Multiple license in serviceHelper.py

### ISSUE THREAD

**Issue by buhtz** [bit-team/backintime#1986]
# Introduction
The issue is most for my own documentation of the process. Short description of the problem: The [`qt/serviceHelper.py`](https://github.com/bit-team/backintime/blob/dev/qt/serviceHelper.py) does have 3 licenses and it is not clear which lines belong to which license. 
I have feedback from a Debian Developer that this make things harder for them.

The **problem** is not one of the licenses itself but that they are **mixed** in the same file and we are not able to distinguish the code between the different licenses.

It is also in question if the MIT license (RedHat and Collabora as copyright holders) related to python-dbus-docs code snippets, is correct used in that file and even relevant.

## History and details about `serviceHelper.py`
- Oldest available version in Launchpad's bazar repo from 2025-01-26. Because I am not familiar with Launchpad and bazar their might be much older versions somewhere: https://bazaar.launchpad.net/~bit-team/backintime/main/revision/1028/qt4/serviceHelper.py#qt4/serviceHelper.py
- The same version can be found in our git repo: https://github.com/bit-team/backintime/commit/edaa248991d7b777eaeea1776e47d534c2b8558a
- I don't know if there is an older version of this file somewhere else.

# MIT / RedHat / Collabora
About the MIT license part related to `dbus-python` ([`python-dbus-docs`](https://dbus.freedesktop.org/doc/dbus-python/tutorial.html)). These are code snippets from the docu which is [licensed as MIT](https://dbus.freedesktop.org/doc/dbus-python/tutorial.html#license-for-this-document). It is unclear if the code snippets themself are covered by this license.

My first approach was to hope to get a clear statement from that project that we can use that snippets and re-license them as GPL-2.0-or-later.

But after some nights of sleep and further line-by-line analysis (see next comment) I can't identify code lines related to python-dbus-doc code snippets.

- I opened a [ticket in September 2024 (refreshed in 4th February 2025)](https://gitlab.freedesktop.org/dbus/dbus-python/-/issues/52) without response.
- I contacted the author (Simon McVittie) directly on 29th Dec via his work email at [collabora](https://www.collabora.com/).
- LinkedIn contact is pending.
- [Gome-GitLab](https://gitlab.gnome.org/smcv) (not contacted yet)
- Announced contact request via our Mastodon account (4th February 2025) -> Success one day later.

Simon McVittie replied and confirmed he got the message and put it on his todo list. But he seems not to be subscribed to this issue.

# "jockey" and GPL2-???

- Some code comments state that parts of the code are from [`jockey`](https://git.launchpad.net/ubuntu/+source/jockey/) project.
- A BIT user found very similar code fragments in a jockey-commit from year 2008 (#2267). I contacted the author of that commit (Martin Pitt via piware.de Domain) and ask for validation and clear license statement.

# CC0 / Public domain
To my understanding it is no problem to re-license public domain code even if we are not the authors.

# Summary
If we could get rid of the MIT license, we can re-license the whole file as GPL-2.0-or-later and thats it.

# Header of `serviceHelper.py`
https://github.com/bit-team/backintime/blob/7f96e5e3fab7375fa8db9ff292c6f8315edf7cea/qt/serviceHelper.py#L1-L57

**Comment by buhtz** [comment]
I did some further line-by-line analysis of the code in serviceHelper.py and from the python-dbus docs. The goal was to identify python-dbus-docs related lines and find out if I can separate them into another file. **But I was not able to find similarities between that two sources!** Therefor I see no reason why the MIT license with RedHat and Collabora as copyright holders was added here.

I cautiously assume that the original author (@germar ?) was a bit too enthusiastic with the licenses and didn’t fully think things through.

I might be wrong but I can't help myself and don't see a reason why the MIT/RedHat/Collabora is mentioned in serviceHelper.py. I wonder if the RedHat/Collabora folks have ever seen or touched the content of that file.

**Comment by martinpitt** [comment]
I'm the original author of https://git.launchpad.net/ubuntu/+source/jockey/tree/jockey/backend.py (ugh, a different lifetime! 😅 ). I went through backintime's serviceHelper.py, and confirm that the only Canonical-copyrighted code is `_check_polkit_privilege()`, and honestly there's not much magic about that -- there's not much choice about how to invoke the polkit API, this is just glue, no logic.  I'd happily treat it as "interface, not invention" and trivial enough to not require this careful copyright tracking.

In this day and age, any AI will probably happily regurgitate such glue in the blink of an eye, too..

So TL/DR: Please feel free to drop the Canonical copyright to ease the tracking of this a bit.

**Comment by buhtz** [comment]
The plan is
- Drop the MIT license incl. RedHad and Collabora author ship. After analiyzing the python-dbus documentation and its code-snippets, no evidence was found that there is a connection to serviceHelper.py. Also no valuable response since Sept. 2024 (17 months from now) from involved persons (Simon McVittie).
- Drop the Canonical author ship. (See [comment](https://github.com/bit-team/backintime/issues/1986#issuecomment-3368102826) from "jockey" (aka @martinpitt).
- Drop CC0 license.
- **Re-license the file as GPL2-or-later**.
- Maybe document this somehow in the code document and/or point the the commit history.
